features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineView.kt

```diff
@@ -218,6 +214,41 @@ private fun MessageShieldDialog(state: TimelineState) {
     )
 }
 
+@OptIn(ExperimentalCoroutinesApi::class)
+@Composable
+private fun TimelinePrefetchingHelper(
+    lazyListState: LazyListState,
+    prefetch: () -> Unit,
+) {
+    // We're using snapshot flows for these because using `LaunchedEffect` with `derivedState` doesn't seem to be responsive enough
+    val firstVisibleItemIndexFlow = snapshotFlow {
```

Comment:
This looks wrong, I don't think `snapshotFlow` should be created in the `Composable` scope, but are intended to be created in the CoroutineScope they are used. I think here you are recreating flows each time

Reply:
You're right, it should be used inside the `LaunchedEffect`. Now I'm worried that this was actually working and the tests passed 😓 .